Sources/Holocron/StorageProvider+Default.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  StorageProvider+Default.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+// MARK: Default Coding
+struct Box<T> {
+    let element: T
+}
+
+extension Box: Decodable where T: Decodable { }
```

Comment:
Can box be extended to `Codable` here instead of to both `Decodable` and `Encodable`?
```
extension Box: Codable where T: Codable {}
```

Reply:
I think the reason they're separate is so that you can choose one or the other and not be forced into both.

Reply:
@earlgaspard yep, what he said. Keeping them separate lets us do this:

```
func value<T: Decodable>(for key: Key) throws -> T?
func write<T: Encodable>(_ value: T, for key: Key) throws
```

Instead of requiring both to conform to `Codable`.